src/main/java/org/betonquest/betonquest/api/EventExecutedOnProfileEvent.java

```diff
@@ -0,0 +1,55 @@
+package org.betonquest.betonquest.api;
+
+import lombok.Getter;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.api.profiles.ProfileEvent;
+import org.betonquest.betonquest.id.EventID;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Fired when a BetonQuest event is executed on a {@link Profile}.
+ * Use {@link EventExecutedEvent} for events executed without a {@link Profile}.
+ */
+public class EventExecutedOnProfileEvent extends ProfileEvent {
```

Comment:
I think wo only hove events that are related on a Profile. So i think the naming does not need to represent this